docs/frontend/webpack-source-maps.md

```diff
@@ -6,8 +6,402 @@ which allows a developer to view / debug developer-friendly source code instead
+* Point to an internal-only / protected map server
+
+per our "Option 2" suggestions above.
+
+[`dist/js/app1.js.map`](../../examples/frontend/webpack-source-maps/dist/js/app1.js.map)
+
+```js
+// Prettified and truncated.
+{
+  "version": 3,
+  "sources": [
+    "webpack:///app1.js",
+    "webpack:///webpack/bootstrap 05941fb93e61871b9ba6?c490",
+    "webpack:///./foo.js?60db",
+    "webpack:///./app1.js"
+  ],
+  "names": [
+    "modules",
+    "__webpack_require__",
+    "moduleId",
+    /* ... SNIPPED ... */
+    "querySelector",
+    "innerHTML"
+  ],
+  "mappings": "CAAS,SAAUA,GCInB,QAAAC,/* ... SNIPPED ... */,WAAAH,EAAA",
+  "file": "app1.js",
+  "sourcesContent": [
+    /* ... SNIPPED ... */
+    "module.exports = function (id, msg) {\n  return \"<h1 id=\\\"\" + id + \"\\\">\" + msg + \"</h1>\";\n};\n\n\n\n//////////////////\n// WEBPACK FOOTER\n// ./foo.js\n// module id = 0\n// module chunks = 0 1",
+    "var foo = require(\"./foo\");\n\ndocument.querySelector(\"#content\").innerHTML += foo(\"app1\", \"App 1\");\n\n\n\n//////////////////\n// WEBPACK FOOTER\n// ./app1.js\n// module id = 1\n// module chunks = 1"
+  ],
+  "sourceRoot": ""
+}
+```
+
+This [JSON structure](http://www.html5rocks.com/en/tutorials/developertools/sourcemaps/#toc-anatomy)
+corresponds to the source map [V3 spec](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit?hl=en_US&pli=1&pli=1).
+The essential information in this file includes:
+
+* The names of the individual files.
+* The full original source code of the individual code files.
+* Names of variables used and mappings of production code location to individual
+  source file locations.
+
+With this file available, a browser can download the map file, attach it to the
+execution environment, and developers are now off to the debugging races with
+the original development source!
+
+Once we build these files, we can load the independent applications with a
+standard index page:
+
+[`index.html`](../../examples/frontend/webpack-source-maps/index.html)
+
+```html
+<!DOCTYPE html>
+<html>
+  <body>
+    <div id="content" />
+    <script src="./dist/js/app1.js"></script>
+    <script src="./dist/js/app2.js"></script>
+  </body>
+</html>
+```
+
+And with these steps, both `app1` and `app2` have individualized full source
+maps and offer a fighting chance for developers debugging live in production.
+
+##### Advantages
+
+**Summary**: You **absolutely should** be enabling source maps in production
+and probably in development too. The disadvantages provided below are really
+just the "costs" of doing the right thing.
+
+* **Debug against original source files**: This is huge. Debugging single
+  application bundles is awkward, and debugging large minified code chunks is
+  downright impossible.
+* **Easy in development / open source**: For development time or projects that
+  don't care if the full source is publicly served via source maps, integrating
```

Comment:
As I've now reworded the distinction to be "public vs private apps", I think this hopefully reads more consistently now as is (?)